### ISSUE TITLE
Not Allowed sign is not shown when dragging attachments on Archived rooms

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#24191]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any Archived Room
2. Drag the attachment on The Archived Room

## Expected Result:
The not Allowed sign is shown when dragging attachments on Archived rooms

## Actual Result:
Copy sign is shown when dragging attachments on Archived rooms

## Workaround:
unknown
\
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.50-0
**Reproducible in staging?:** Y  
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://github.com/Expensify/App/assets/43995119/1996f0dd-0a00-4d3f-b4ae-d0fb0256b1ec

https://github.com/Expensify/App/assets/43995119/ec80a070-ceee-4ee9-ab3e-557e1dfb8cb8

**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690837081412469 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by flaviadefaria** [comment]
Added the external label.

**Comment by Santhosh-Sellavel** [comment]
@allroundexperts This seems like a duplicate of https://github.com/Expensify/App/issues/22218 as the problem is the same  This should be handled there as well!

**Comment by noahroman** [comment]
Hi,
I tried a lot on this issue, what I came to know is there are 4 possible values to the drop effect and in case of archived room I have used 'none' for the cursor drop effect. I also passed additional param for checking archive rooms to drag drop component.

Please check this video link and let me know if this work for you
https://www.loom.com/share/17278877c8794e17a3754b98d327538d?sid=f0389e05-e68f-41c5-9e3f-76cc417b2e44

**Comment by noahroman** [comment]
Contributor details
Your Expensify account email: thenoumanmuzammil@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01e4d32778f98872aa

**Comment by noahroman** [comment]
Hi,
I have tried a lot on this issue, to my knowledge i can only find 4 possible values to dropeffect. And In case of archive rooms I have set this to 'none'. I also pass additional param to dragndrop component to check whether it is archive room or not
Please look at this video and share your thoughts

https://www.loom.com/share/17278877c8794e17a3754b98d327538d?sid=967c6a31-bb4c-4478-87bc-4aa9085a2a87

**Comment by noahroman** [comment]
This is how fixed code performs in a window os 

https://github.com/Expensify/App/assets/95413216/4564a938-9e29-4a9f-9ea9-8059a3486aa6

**Comment by Santhosh-Sellavel** [comment]
@noahroman please take your concern here to #22218.

cc: @flaviadefaria this is on hold for #22218

**Comment by noahroman** [comment]
@Santhosh-Sellavel Should I start working to that issue too?

**Comment by Santhosh-Sellavel** [comment]
@flaviadefaria please add hold to the title and remove help wanted label thanks!

**Comment by Santhosh-Sellavel** [comment]
On hold

**Comment by flaviadefaria** [comment]
Still on hold

**Comment by flaviadefaria** [comment]
Still on hold.

**Comment by flaviadefaria** [comment]
Still on hold

**Comment by flaviadefaria** [comment]
@Santhosh-Sellavel https://github.com/Expensify/App/issues/22218 has been closed does this mean that this issue is fixed?

**Comment by flaviadefaria** [comment]
Removed the hold so that we can take action and either close or fix (in case there's something still tobe fixed).

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel #22218 should solve this. We might want to put this one on hold for that.

@allroundexperts Is this solved now?

**Comment by Santhosh-Sellavel** [comment]
Seems it's not solved yet?

https://github.com/Expensify/App/assets/85645967/f14ee4fc-31dd-4a4e-92d4-55a09d90c847

**Comment by flaviadefaria** [comment]
@Santhosh-Sellavel we should still look for proposals here, right?

**Comment by akamefi202** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Not Allowed sign is not shown when dragging attachments on Archived rooms.

### What is the root cause of that problem?
https://github.com/Expensify/App/blob/562f9d8ff66d1e90285df088fa785c9638aa3861/src/pages/home/ReportScreen.js#L412
Currently, we only disallow drag/drop if the report is not ready for display.
So the copy sign is still shown when dragging attachment on archived room.

https://github.com/Expensify/App/blob/562f9d8ff66d1e90285df088fa785c9638aa3861/src/pages/home/report/ReportActionCompose/ReportActionCompose.js#L412-L420
Although drag/drop is allowed, we can't drop file on archived room anyway.
Because archived room doesn't have the composer, which means the report screen doesn't have `ReportDropUI` component to handle drop event.

As a result, we can't drop file on archived room but copy sign is shown while dragging attachment.

### What changes do you think we should make in order to solve the problem?
```
<DragAndDropProvider isDisabled={!isReportReadyForDisplay || ReportUtils.isArchivedRoom(report)}>
```
We should disallow drag/drop in the report screen if the report is archived.
As a result, the not-allowed sign will be shown when dragging attachment on archived room.

### What alternative solutions did you explore? (Optional)
N/A

**Comment by Santhosh-Sellavel** [comment]
> From [#24191 (comment)](https://github.com/Expensify/App/issues/24191#issuecomment-1674165678)
> 
> > @Santhosh-Sellavel #22218 should solve this. We might want to put this one on hold for that.
> 
> @allroundexperts Is this solved now?

Bump @allroundexpertsm how did this one get missed?

**Comment by Santhosh-Sellavel** [comment]
> [22218](https://github.com/Expensify/App/issues/22218) is for disabling drag and drop (show disallow cursor) by default

Where do we show a disabled cursor, I don't see any disabled cursor at all. I just see an Arrow cursor is that what is expected at the other PR?

**Comment by akamefi202** [comment]
@Santhosh-Sellavel Are you testing on Mac? It doesn't show a disabled cursor on Mac. But on Windows, it shows disabled cursor.

**Comment by Santhosh-Sellavel** [comment]
Yeah, I'm testing on mac!

**Comment by flaviadefaria** [comment]
@garrettmknight  I'm officially no longer in the BugZero team so reassigning this as it will not close until tomorrow.

**Comment by garrettmknight** [comment]
@Santhosh-Sellavel just catching up on this one. Is this a good summary of where we're at?
- This was on hold for another fix
- The other fix was finally completed
- This issue still exists, but only on Macs
- We're waiting for proposals

**Comment by Santhosh-Sellavel** [comment]
Yeah pretty much, 
> This issue still exists, but only on Macs

This exists on both platforms. But confused about the other issue fix which is expected to show a disabled cursor on mac but it's not!

Waiting for @allroundexperts to answer this or @ginsuma 
> > [22218](https://github.com/Expensify/App/issues/22218) is for disabling drag and drop (show disallow cursor) by default
> 
> Where do we show a disabled cursor, I don't see any disabled cursor at all. I just see an Arrow cursor is that what is expected at the other PR?

bump @ginsuma or @allroundexperts

**Comment by garrettmknight** [comment]
So, if we were to fix this, we'd likely be adding a fix for MacOS only. Is that right @Santhosh-Sellavel ?

I noticed there isn't any error thrown so if you do this on a mac you're just dropping without any failure indication.

**Comment by Santhosh-Sellavel** [comment]
@hungvu193's proposal [here](https://github.com/Expensify/App/issues/24191#issuecomment-1667253937) looks good to me!
C+ Reviewed
🎀 👀 🎀 

@garrettmknight 
> So, if we were to fix this, we'd likely be adding a fix for MacOS only. Is that right @Santhosh-Sellavel ?

Not exactly, seems platforms handle things in their own way. Seems the current behaviour with Mac seems to be expected as per this [comment](https://github.com/Expensify/App/issues/24191#issuecomment-1765653432). So we will be consistent and address this missing case too!

**Comment by Santhosh-Sellavel** [comment]
Deployed to production last [week](https://github.com/Expensify/App/pull/30229#issuecomment-1811402272).

**Comment by garrettmknight** [comment]
**Summary of payments:**
- Reporter: @daveSeife $50 paid via upwork
- Contributor: @hungvu193 $1000 paid via upwork
- Reviewer: @Santhosh-Sellavel $1000

Upwork job: https://www.upwork.com/jobs/~01340b7f14e5ffd467

**Comment by garrettmknight** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 

- [ ] [@Santhosh-Sellavel ] The PR that introduced the bug has been identified. Link to the PR:
- [ ]  [@Santhosh-Sellavel ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ]  [@Santhosh-Sellavel ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ]  [@Santhosh-Sellavel ] Determine if we should create a regression test for this bug.
- [ ]  [@Santhosh-Sellavel ] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
- [ ]  [@garrettmknight] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by garrettmknight** [comment]
@Santhosh-Sellavel can you complete the checklist and request payment?

**Comment by daveSeife** [comment]
.@garrettmknight The reporter's payment should be $250. This matter was raised prior to the deduction occurring.

**Comment by garrettmknight** [comment]
@daveSeife fair - updated payment.

**Comment by daveSeife** [comment]
I have Received the payment. Thank You! @garrettmknight

**Comment by Santhosh-Sellavel** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@Santhosh-Sellavel ] The PR that introduced the bug has been identified. Link to the PR: N/A It was there right from 
* [x]   [@Santhosh-Sellavel ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A 
* [x]   [@Santhosh-Sellavel ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
* [x]   [@Santhosh-Sellavel ] Determine if we should create a regression test for this bug. N/A Minor UI
* [x]   [@Santhosh-Sellavel ] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
* [x]   [@garrettmknight] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by JmillsExpensify** [comment]
$1,000 payment approved for @Santhosh-Sellavel based on [this comment](https://github.com/Expensify/App/issues/24191#issuecomment-1821627518).
